oeps/oep-0005/decisions/0004-To-Make-Or-Not-To-Make.rst

```diff
@@ -0,0 +1,65 @@
+To Make or Not to Make
+======================
+
+
+Status
+------
+
+Accepted
+
+
+Context
+-------
+
+For Decentralized Devstack, an common interface is necessary to enable a OpenedX Developer to add contributions to the many independent services in Open edX codebase.
+
+Options
+~~~~~~~
+
+1. The current Devstack uses make targets as an basic interface for developers.
+
+2. Use docker-compose commands directly
+
+Case for and against make targets
+`````````````````````````````````
+
+For:
+
+- This standard already exists and is known by developers at edx.
+
+  * If we don't create a make interface now, sevices might create their on make interface, which might/will result in non-standard make experience
+- make targets decrease verbosity and memorization
+
+Against:
+
+- make targets do not provide much value
+
+  * most targets wrap commands that are/can be only few characters longer
+
+- by wrapping, make targets mask what is happening, thus adding another layer to disentangle during debugging
+- comes with maintanence costs
```

Comment:
Here, too, I would reiterate that there may still be makefiles that need to be maintained, but removing process control from them would lower their footprint.

Reply:
> Here, too, I would reiterate that there may still be makefiles that need to be maintained, but removing process control from them would lower their footprint.

Agreed, except I've never heard the term "process control", although I can infer what you mean by it. How about "managing images and containers"?

Reply:
Might be an old-school term for starting, stopping, etc. a process; "managing images and containers" may be more accurate here.

Reply:
On second thought, "managing images and containers" may be accurate for Docker-based projects; if we move everything to Docker then that's fine.

However, if we have projects that are necessary for a working development environment that are _not_ Docker-based, we need to describe the basic operations, such as launching, in a more general way.

Again, improving our use of Docker is important, but what we really, desperately, need is a much better way to manage our working development environment.